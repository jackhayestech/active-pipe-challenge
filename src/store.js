import { createStore, applyMiddleware, compose } from 'redux';
import createRootReducer from './redux/reducers/Root.reducer';

/*
*
*/
const middlewareCompose = compose(
  applyMiddleware(),
);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  createRootReducer(),
  composeEnhancers(middlewareCompose),
);

export default store;
