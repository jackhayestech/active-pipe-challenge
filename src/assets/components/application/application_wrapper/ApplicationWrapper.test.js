import React from 'react';
import { create } from "react-test-renderer";
import { Provider } from 'react-redux';
import store from '../../../../store';
import ApplicationWrapper from './ApplicationWrapper.component';

const testProp = [{
  "id": 1,
  "status": "current",
  "street": "1 Fake Street",
  "suburb": "Glen Iris",
  "state": "VIC",
  "postcode": "3146",
  "price": 1230500,
  "image": "https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
},];

describe("The wrapper for the application", () => {
  test("No properties displayed", () => {
    const component = create(
      <Provider store={store}>
        <ApplicationWrapper />
      </Provider>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("No properties found", () => {
    const component = create(
      <Provider store={store}>
        <ApplicationWrapper properties={[]}/>
      </Provider>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});