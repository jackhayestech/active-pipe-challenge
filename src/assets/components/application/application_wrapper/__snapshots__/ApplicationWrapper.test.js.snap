// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The message alert item To display error message 1`] = `null`;

exports[`The wrapper for the application To display error message 1`] = `null`;
